src/Docker.DotNet/DockerClient.cs

```diff
@@ -112,6 +112,22 @@ await sock.ConnectAsync(new Microsoft.Net.Http.Client.UnixDomainSocketEndPoint(p
                     uri = new UriBuilder("http", uri.Segments.Last()).Uri;
                     break;
 
+                case "ssh":
```

Comment:
I'm waiting for this PR, but I have to say that this is a bad design. This code violates an O principle from SOLID (Open-Close). Every time someone wants to add a new protocol, even if it's implemented in a separate library, we will have to change this code.

Maybe we should change the constructor:

```diff
- internal DockerClient(DockerClientConfiguration configuration, Version requestedApiVersion)
+ internal DockerClient(DockerClientConfiguration configuration, Version requestedApiVersion, ManagedHandler handler, Uri endpointBaseUri) 
```
And entire code from https://github.com/dotnet/Docker.DotNet/blob/92a9557039585e55a535bc3e330bd06dff3720d5/src/Docker.DotNet/DockerClient.cs#L50-L133 move to the `DockerClientConfiguration` class. 
And then in the external library `Dotnet.Docker.SshClient` or something like that there would be possible to create a configuration inheriting from `DockerClientConfiguration`:

```C# 
public class SshDockerClientConfiguraion: DockerClientConfiguration
{
//pseudo code:
    public override ManageHandler CreateHandler(URI url)
    {
        if (url.schema is wrong) { error }
        return GodImplementation
    }

    public override URI GetEndpointUri(URL url) {}
}
```

There will likely be more challenges and this can be solved in other ways. This is only a proposed solution.

Reply:
I've implemented that, good suggestion